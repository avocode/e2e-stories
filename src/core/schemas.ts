import Joi from "joi"
import { SchemasI } from './types'

export default new class Schemas implements SchemasI {
    stepsSchema = Joi.object().keys({
        component: Joi.string().min(1),
        visit: Joi.string().min(5),
        click: Joi.string().min(1),
        dblClick: Joi.string().min(1),
        urlIs: Joi.string().min(1),
        exists: Joi.string().min(1),
        notExists: Joi.string().min(1),
        pause: Joi.number().integer().min(0),
        fill: Joi.object().keys({
            el: Joi.string().min(1).required(),
            text: Joi.string().min(1).required(),
        }),
        textIs: Joi.object().keys({
            el: Joi.string().min(1).required(),
            text: Joi.string().min(1).required(),
        }),
        select: Joi.object().keys({
            el: Joi.string().min(1).required(),
            text: Joi.string().min(1).required(),
        }),
        upload: Joi.object().keys({
            el: Joi.string().min(1).required(),
            file: Joi.string().min(1).required(),
        }),
        keyPress: Joi.string().min(1),
        screenshot: Joi.object().keys({
            path: Joi.string().required(),
            type: Joi.string().min(3).max(4),
            fullPage: Joi.boolean(),
        })
    })
  
    schema = Joi.object().keys({
        name: Joi.string().min(10).required(),
        only: Joi.bool(),
        skip: Joi.bool(),
        todo: Joi.string(),
        testSettings: Joi.object().keys({
            retryTimes: Joi.number().min(1),
            setTimeout: Joi.number().min(2),
        }),
        steps: Joi.array().items(this.stepsSchema).min(1).required(),
    })
  
    schemaWithoutName = Joi.object().keys({
        name: Joi.string().strip(),
        steps: Joi.array().items(this.stepsSchema).min(1).required(),
    })
}