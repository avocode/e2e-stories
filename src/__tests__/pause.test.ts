import Core from "../templates/puppeteer/core"
import schemas from "../core/schemas"
import fs from 'fs'
import path from "path"

it("Converts yaml pause command to puppeteer testing code", async () => {
  const convertor = new Core()

  const filePath = path.resolve(path.join(__dirname, "/stories/"), "pause.tst.yaml")

  const rawYaml = fs.readFileSync(filePath, "utf-8")

  const result = convertor.convertYamlToJs(rawYaml, schemas.schema, filePath)
    
  expect(result).toMatchSnapshot()
})