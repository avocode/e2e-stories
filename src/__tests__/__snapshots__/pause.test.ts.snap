// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converts yaml pause command to puppeteer testing code 1`] = `
"describe(\`Pauses a test for specific amount of time (in ms)\`, function() {
	let context
	let random
	let page

	beforeEach(async () => {
		context = await browser.createIncognitoBrowserContext()
		page = await context.newPage()
		random = +new Date
		page.setExtraHTTPHeaders({})

		return true
	})

	afterEach(async () => {
		await page.close()
		return true
	})

	jest.retryTimes(3)
	jest.setTimeout(50000)
	it(\`Pauses a test for specific amount of time (in ms)\`, async () => {
		await page.waitForTimeout(5000)
		return true
	})
})"
`;
